tests/Text/URISpec.hs

```diff
@@ -263,3 +283,62 @@ shouldParseBs s a =
       "the parser is expected to succeed, but it failed with:\n" ++
       parseErrorPretty' s e
     Right a' -> a' `shouldBe` a
+
+
+-- | Test cases from section 5.4.1 from RFC 3986
+--
+-- First item in the tuple is the relative path, the second is the expected
+-- result. The base path is always 'http://a/b/c/d;p?q'.
+--
+testCasesRFC3986 :: Bool -- ^ Strict or non-strict
+                 -> [(T.Text, T.Text)]
+testCasesRFC3986 strict = [
```

Comment:
I'd drop `strict` here because it only changes one test case, which could be annotated and tested separately and more explicitly.